packages/openmrs/src/Adaptor.js

```diff
@@ -40,252 +37,375 @@ export function execute(...operations) {
  * Logs in to OpenMRS, gets a session token.
  * @example
  *  login(state)
- * @function
+ * @private
  * @param {State} state - Runtime state.
  * @returns {State}
  */
 function login(state) {
   const { instanceUrl, username, password } = state.configuration;
 
-  const params = {
-    method: 'GET',
-    url: `${instanceUrl}/ws/rest/v1/session`,
-    auth: { username, password },
-    jar: true,
-  };
-
+  const agent = request.agent();
   return new Promise((resolve, reject) => {
-    request(params, function (error, response, body) {
-      error = assembleError({ error, response, params });
-      if (error) {
-        reject(error);
-      } else {
-        const resp = tryJson(body);
-        resolve({ ...state, auth: resp });
-      }
-    });
+    agent
+      .get(`${instanceUrl}/ws/rest/v1/session`)
+      .auth(username, password)
+      .then(() => {
+        resolve({ ...state, agent });
+      });
```

Comment:
I'm not sure what is going on here, it doesn't need to be wrapped in a promise; and you don't seem to be getting the response from the server?